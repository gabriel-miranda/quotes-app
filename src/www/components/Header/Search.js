import React from 'react';
import PropTypes from 'prop-types';

const INPUT_STYLES = {
  background: '#e9eaec',
  border: 'none',
  padding: '8px 25px 8px 32px',
  boxShadow: 'none',
  height: '35px',
};

const CLOSED_INPUT = {
  width: '10px',
  padding: '8px 19px',
};

const CLOSED_STYLES = {
  ...INPUT_STYLES,
  ...CLOSED_INPUT,
};

const LABEL_STYLE = {
  position: 'relative',
  margin: '6px 0',
  float: 'right',
};

const ICON_STYLES = {
  position: 'absolute',
  top: '6px',
  left: '12px',
};

const OPEN_FORM_STYLES = {
  position: 'absolute',
  top: '-20px',
  bottom: '-20px',
  width: 'calc(100% - 30px)',
  height: 'calc(100% + 34px)',
  background: '#fff',
  padding: '20px 0',
  zIndex: '1',
};

const FORM_STYLES = {
  float: 'right',
};

const RADIO_STYLES = {
  margin: '0 .5em 0 0',
};

const RADIO_LABEL_STYLES = {
  padding: '12px 8px',
  fontSize: '12px',
};

const CROSS_STYLES = {
  position: 'absolute',
  top: '6px',
  right: '10px',
  cursor: 'pointer',
};

const RadioButton = ({
  label,
  value,
  checked,
  onChange,
}) => (
  <label style={RADIO_LABEL_STYLES} htmlFor={value}>
    <input
      type="radio"
      value={value}
      checked={checked}
      onChange={onChange}
      style={RADIO_STYLES}
      name={value}
    />
    {label}
  </label>
);

RadioButton.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default class Search extends React.Component {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
  };

  state = {
    open: false,
    text: '',
    searchBy: 'authorName',
  };

  setRef = (ref) => {
    this._input = ref;
  }

  handleFocus = () => {
    this.setState({open: true});
  }

  handleBlur = () => {
    this._input.blur();
    this.setState({open: false});
  }

  handleOnChange = (e) => {
    const text = e.target.value;
    this.setState({text});
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { searchBy, text } = this.state;
    this.handleBlur();
    this.props.onSubmit(searchBy, text);
  }

  handleSearchBy = (e) => {
    const searchBy = e.target.value;
    this.setState({searchBy});
  }

  isChecked(label) {
    return label === this.state.searchBy;
  }

  render() {
    const { open } = this.state;
    return (
      <form
        onSubmit={this.handleSubmit}
        style={open ? OPEN_FORM_STYLES : FORM_STYLES}
      >
        <label style={LABEL_STYLE} htmlFor="search">
          <i
            className="icon-budicon-489 icon"
            style={ICON_STYLES}
            onClick={this.handleFocus}
          />
          {open &&
            <i
              className="icon-budicon-471 icon"
              style={CROSS_STYLES}
              onClick={this.handleBlur}
            />
          }
          <input
            className="form-control"
            style={open ? INPUT_STYLES : CLOSED_STYLES}
            onChange={this.handleOnChange}
            onFocus={this.handleFocus}
            value={this.state.text}
            name="search"
            placeholder="Search here..."
            ref={this.setRef}
          />
        </label>
        { open &&
          <div>
            <RadioButton
              label="Text"
              value="text"
              checked={this.isChecked('text')}
              onChange={this.handleSearchBy}
            />
            <RadioButton
              label="Author"
              value="authorName"
              checked={this.isChecked('authorName')}
              onChange={this.handleSearchBy}
            />
          </div>
        }
      </form>
    );
  }
}
