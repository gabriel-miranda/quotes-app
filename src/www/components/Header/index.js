import React from 'react';
import PropTypes from 'prop-types';
import Container from '../Layout/Container';
import Content from '../Layout/Content';
import Search from './Search';

const HEADER_STYLES = {
  background: '#fff',
};

const QUOTES_LOGO_STYLES = {
  display: 'flex',
  alignItems: 'center',
};

const LOGO_STYLES = {
  height: '42px',
  margin: '0 1em 0 0',
};

const LOGO_TITLE_STYLES = {
  margin: '0',
  fontSize: '25px',
};

const LOGIN_WRAPPER_STYLE = {
  padding: '5px',
  float: 'right',
};

const LOGIN_BUTTON_STYLE = {
  fontSize: '11px',
  color: '#fff',
  padding: '7px 14px',
};

const Header = ({handleSearch}) => (
  <header className="site-header" style={HEADER_STYLES}>
    <nav className="navbar navbar-default">
      <Container>
        <Content>
          <div className="navbar-header pull-left">
            <div style={QUOTES_LOGO_STYLES}>
              <img
                src="/static/logo.png"
                alt="quotes app logo"
                style={LOGO_STYLES}
              />
              <h1 style={LOGO_TITLE_STYLES}>
                <span className="hidden-xs">
                  Quotes app
                </span>
                <span className="hidden-sm hidden-md hidden-lg">
                  QA
                </span>
              </h1>
            </div>
          </div>
          <Search onSubmit={handleSearch} />
          <div style={LOGIN_WRAPPER_STYLE}>
            <a
              className="btn btn-success"
              style={LOGIN_BUTTON_STYLE}
            >
              Login
            </a>
          </div>
        </Content>
      </Container>
    </nav>
  </header>
);

Header.propTypes = {
  handleSearch: PropTypes.func.isRequired,
};

export default Header;
