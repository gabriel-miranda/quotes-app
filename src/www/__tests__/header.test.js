/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../components/Header';

const dummyFn = () => {};

describe('Header test with enzyme', () => {
  it('should show header title correctly on small and up devices"', () => {
    const header = shallow(<Header handleSearch={dummyFn} />);
    expect(header.find('h1').find('.hidden-xs').text()).toEqual('Quotes app');
  });

  it('should show header title correctly on extra small devices"', () => {
    const header = shallow(<Header handleSearch={dummyFn} />);
    expect(header.find('h1').find('.hidden-sm').text()).toEqual('QA');
  });
});

describe('Header test with Snapshot Testing', () => {
  it('should show header title correctly', () => {
    const component = renderer.create(<Header handleSearch={dummyFn} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
