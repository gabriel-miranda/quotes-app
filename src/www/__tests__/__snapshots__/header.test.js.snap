// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test with Snapshot Testing should show header title correctly 1`] = `
<header
  className="site-header"
  style={
    Object {
      "background": "#fff",
    }
  }
>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 col-md-offset-3 clearfix"
        >
          <div
            className="navbar-header pull-left"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <img
                alt="quotes app logo"
                src="/static/logo.png"
                style={
                  Object {
                    "height": "42px",
                    "margin": "0 1em 0 0",
                  }
                }
              />
              <h1
                style={
                  Object {
                    "fontSize": "25px",
                    "margin": "0",
                  }
                }
              >
                <span
                  className="hidden-xs"
                >
                  Quotes app
                </span>
                <span
                  className="hidden-sm hidden-md hidden-lg"
                >
                  QA
                </span>
              </h1>
            </div>
          </div>
          <form
            onSubmit={[Function]}
            style={
              Object {
                "float": "right",
              }
            }
          >
            <label
              htmlFor="search"
              style={
                Object {
                  "float": "right",
                  "margin": "6px 0",
                  "position": "relative",
                }
              }
            >
              <i
                className="icon-budicon-489 icon"
                onClick={[Function]}
                style={
                  Object {
                    "left": "12px",
                    "position": "absolute",
                    "top": "6px",
                  }
                }
              />
              <input
                className="form-control"
                name="search"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search here..."
                style={
                  Object {
                    "background": "#e9eaec",
                    "border": "none",
                    "boxShadow": "none",
                    "height": "35px",
                    "padding": "8px 19px",
                    "width": "10px",
                  }
                }
                value=""
              />
            </label>
          </form>
          <div
            style={
              Object {
                "float": "right",
                "padding": "5px",
              }
            }
          >
            <a
              className="btn btn-success"
              style={
                Object {
                  "color": "#fff",
                  "fontSize": "11px",
                  "padding": "7px 14px",
                }
              }
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
